test/unit/background/datastore-test.js

```diff
@@ -0,0 +1,266 @@
+      httpRealm: null,
+      username: "QUUXuser",
+      password: "QUUXpass",
+      usernameField: "username",
+      passwordField: "password",
+    };
+    const item = convertInfo2Item(info);
+
+    // TODO: Issue #21 should do away with item/info conversion
+    const addedItem = await store.add(item);
+    const expectedItem = {
+      ...item,
+      id: addedItem.id,
+    };
+    expect(addedItem).to.deep.equal(expectedItem);
+
+    const resultItem = await store.get(addedItem.id);
+    expect(resultItem).to.deep.equal(addedItem);
+
+    const apiAdd = browser.experiments.logins.add;
+    expect(apiAdd.callCount).to.equal(1);
+
+    const expectedApiInfo = {
+      // TODO: Issue #21 should do away with item/info conversion
+      // A double conversion - semi-reflects what happens.
+      ...convertItem2Info(item),
+      guid: addedItem.id,
+    };
+
+    const {
+      timesUsed,
+      timeLastUsed,
+      timeCreated,
+      timePasswordChanged,
+      ...resultApiInfo
+    } = apiAdd.lastCall.lastArg;
+
+    expect(timesUsed).to.equal(0);
+    expect(timeLastUsed).to.not.be.undefined;
+    expect(timeCreated).to.not.be.undefined;
+    expect(timePasswordChanged).to.not.be.undefined;
+    expect(resultApiInfo).to.deep.equal(expectedApiInfo);
+  });
+
+  it("allows an item to be updated", async () => {
+    const id = "BAR";
+
+    const originalItem = await store.get(id);
+    const expectedItem = {
+      ...originalItem,
+      entry: {
+        ...originalItem.entry,
+        password: "updated password",
+      },
+    };
+
+    const updatedItem = await store.update(expectedItem);
+    expect(updatedItem).to.deep.equal(expectedItem);
+
+    const fetchedItem = await store.get(id);
+    expect(fetchedItem).to.deep.equal(expectedItem);
+
+    const apiUpdate = browser.experiments.logins.update;
+    expect(apiUpdate.callCount).to.equal(1);
+
+    const expectedApiInfo = {
+      // TODO: Issue #21 should do away with item/info conversion
+      // A double conversion - semi-reflects what happens.
+      ...convertItem2Info(expectedItem),
+      guid: expectedItem.id,
+    };
+
+    const {
+      timePasswordChanged,
+      ...resultApiInfo
+    } = apiUpdate.lastCall.lastArg;
+
+    expect(timePasswordChanged).to.not.be.undefined;
+    expect(resultApiInfo).to.deep.equal(expectedApiInfo);
+  });
+
+  it("allows an item to be removed", async () => {
+    const id = "BAZ";
+
+    const beforeItem = await store.get(id);
+    expect(beforeItem).to.not.equal(null);
+
+    const removedItem = await store.remove(id);
+    expect(removedItem).to.deep.equal(beforeItem);
+
+    const afterItem = await store.get(id);
+    expect(afterItem).to.equal(null);
+
+    const apiRemove = browser.experiments.logins.remove;
+    expect(apiRemove.callCount).to.equal(1);
+    expect(apiRemove.lastCall.lastArg).to.equal(id);
+  });
+});
+
+const cmpAlphaBy = name => (a, b) => a[name].localeCompare(b[name]);
+
+const SAMPLE_INFOS = {
```

Comment:
For completeness' sake, probably worth including the rest of the [`nsILoginMetaInfo`](https://searchfox.org/mozilla-central/source/toolkit/components/passwordmgr/nsILoginMetaInfo.idl) fields in these mocks: `timeCreated`, `timeLastUsed`, `timePasswordChanged`, `timesUsed`.